packages/storage/src/providers/axios-http-handler.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+import {
+	BrowserHttpOptions,
+	HttpHandler,
+	HttpHandlerOptions,
+	HttpRequest,
+	HttpResponse,
+} from '@aws-sdk/types';
+import { buildQueryString } from '@aws-sdk/querystring-builder';
+import axios, { AxiosRequestConfig, Method } from 'axios';
+import { ConsoleLogger as Logger } from '@aws-amplify/core';
+
+const logger = new Logger('axios-http-handler');
+
+export class AxiosHttpHandler implements HttpHandler<Blob, BrowserHttpOptions> {
+	constructor(
+		private readonly httpOptions: BrowserHttpOptions = {},
+		private readonly emitter?: any
+	) {}
+
+	destroy(): void {
+		// Do nothing. TLS and HTTP/2 connection pooling is handled by the
+		// browser.
+	}
+
+	handle(
+		request: HttpRequest<Blob>,
+		options: HttpHandlerOptions
+	): Promise<HttpResponse<Blob>> {
+		const requestTimeoutInMs = this.httpOptions.requestTimeout;
+		const emitter = this.emitter;
+
+		let path = request.path;
+		if (request.query) {
+			const queryString = buildQueryString(request.query);
+			if (queryString) {
+				path += `?${queryString}`;
+			}
+		}
+
+		const port = request.port;
+		const url = `${request.protocol}//${request.hostname}${

```

Comment:
I believe the ":" is missing? 

```suggestion
		const url = `${request.protocol}://${request.hostname}${
```

Reply:
This code is taken directly from aws-sdk's fetch implementation. I'm assuming that `:` is part of the protocol.